mlir/lib/Conversion/ArithToEmitC/ArithToEmitC.cpp

```diff
@@ -112,6 +112,78 @@ class CmpIOpConversion : public OpConversionPattern<arith::CmpIOp> {
   }
 };
 
+template <typename ArithOp, bool needsUnsigned>
+class CastConversion : public OpConversionPattern<ArithOp> {
+public:
+  using OpConversionPattern<ArithOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ArithOp op, typename ArithOp::Adaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Type opReturnType = this->getTypeConverter()->convertType(op.getType());
+    if (!isa_and_nonnull<IntegerType>(opReturnType)) {
+      return rewriter.notifyMatchFailure(op, "expected integer result type");
+    }
+
+    if (adaptor.getOperands().size() != 1) {
+      return rewriter.notifyMatchFailure(
+          op, "CastConversion only supports unary ops");
+    }
```

Comment:
```suggestion
    if (adaptor.getOperands().size() != 1)
      return rewriter.notifyMatchFailure(
          op, "CastConversion only supports unary ops");
```

Reply:
For multi-line statements like this one, I was [asked to put braces](https://github.com/Xilinx/llvm-project/pull/177#discussion_r1593501937). The LLVM [coding standards](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements) do not give out rules for this specific case, but I'd like to make up my mind -- is there an LLVM officially accepted practice on this?

Reply:
I don't have strong feelings about that, it seems to be borderline for readability, so I am fine with keeping the braces.